service/storage_proxy.cc

```diff
@@ -6932,7 +6932,7 @@ future<> storage_proxy::abort_view_writes() {
 
 future<>
 storage_proxy::stop() {
-    return make_ready_future<>();
+    return _pending_writes_phaser.close();
```

Comment:
I am not sure what difference this makes since IIRC this function is never called.

Reply:
It's called in cql_test_env, but indeed not called in main.
I'm actually testing right a now a patch to call proxy.stop() in main as well.

Reply:
Added to v5 ([9afde52](https://github.com/scylladb/scylladb/commit/9afde5277be7ef88dfab45e07026ca39872909b5))

Reply:
>  I'm actually testing right a now a patch to call proxy.stop() in main as well.

Here be dragons

Reply:
Added an async gate to query processor held around queries and closed on stop
plus added a call to qp.stop in main - this prevents use of storage proxy via the query processor after the proxy is stopped (and shards destroyed).